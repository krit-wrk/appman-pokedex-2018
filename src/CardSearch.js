import React from 'react'
import { Row, Col, Box } from './Layout'
import {Card} from './Card'
export function CardSearch(props) {
	const { myDex, cards } = props;

	const cardfiltered = cards.filter(c => {
		var f = myDex.find(m => m.id == c.id)
		if (f) return false
		else return true
	})
	return (
		<div style={{
			backgroundColor: "#FFFFFF",
			width: "700px",
			height: "100%",
			overflowY: "scroll"
		}}>
			<Row>
				<label>Name</label> <input onChange={({target:{value}})=>props.inputChange("searchName",value)} value={props.searchName}/>
				<label>Type</label> <input onChange={({target:{value}})=>props.inputChange("searchType",value)} value={props.searchType}/>
				<button onClick={props.searchCard}>Search</button>
			</Row>
			{
				cardfiltered.map(c => <Card key={c.id} card={c} action={props.addCard} btnDisplay="Add" />)
			}
		</div>
	)
}


