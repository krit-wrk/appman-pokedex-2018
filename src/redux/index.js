export { default as store } from "./store";
export { default as myDexAction } from "./myDex/myDexAction";
