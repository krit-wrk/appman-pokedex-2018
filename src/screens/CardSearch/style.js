import { StyleSheet } from "aphrodite";
import { COLORS } from "../../constants";
export default StyleSheet.create({
	container: {
		display: "flex",
		width: "900px",
		margin: "32px",
		flex: 1,
		flexDirection: "column",
		backgroundColor: COLORS.ModalContentBackground,
	},
	titleContainer: {
		display: "flex",
		flexDirection: "row",
		alignItems: "center",
		justifyContent: "center",
		borderColor: COLORS.SearchBoxBorder,
		borderStyle: "solid",
		padding: "8px",
		margin: "16px",
	},
	seacrhInput: {
		fontSize: "40px",
		borderStyle: "none",
		paddingLeft: "8px",
		flex: 1,
		fontFamily: "Gaegu",
	},
	seacrhImg: {
		width: "50px",
	},
	cardListContainer: {
		display: "flex",
		flex: 1,
		flexDirection: "column",
		overflowX: "scroll",
		alignItems: "center",
	},
	EmptyContainer: {
		flex: 1,
		display: "flex",
		flexDirection: "column",
		justifyContent: "center",
		alignIems: "center",
	},
	EmptyText: {
		textAlign: "center",
		fontSize: "50px",
		fontFamily: "Gaegu",
		color: COLORS.EmptyTextColor,
	},
	card: {
		width: "95%",
		padding: "16px",
	},

	addButton: {
		textAlign: "center",
		fontSize: "25px",
		color: COLORS.AddButton,
	},
});
