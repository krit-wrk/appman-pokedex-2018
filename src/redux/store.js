import { createStore, combineReducers } from "redux";

import myDexReducer from "./myDex/myDexReducer";

const rootReducer = combineReducers({
	myDex: myDexReducer,
});

const store = createStore(rootReducer);

export default store;
