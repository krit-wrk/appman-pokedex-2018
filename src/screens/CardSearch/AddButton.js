import React from "react";
import { css } from "aphrodite";
import style from "./style";

export const AddButton = ({ onClick }) => (
	<div className={css(style.addButton)} onClick={onClick}>
		Add
	</div>
);
