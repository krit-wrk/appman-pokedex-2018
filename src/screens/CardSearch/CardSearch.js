import React, { useMemo } from "react";
import { Card } from "../../components";
import style from "./style";
import { css } from "aphrodite";
import { COLORS } from "../../constants";
import search from "./../../assets/search.png";
const CardSearch = () => {
	const list = useMemo(() => {
		let l = [];
		for (let i = 0; i < 100; i++) {
			l.push(
				<div className={css(style.card)}>
					<Card
						name="Deoxys ex"
						happy={3}
						hp={30}
						image={"https://images.pokemontcg.io/ex8/98.png"}
						str={40}
						weak={90}
					/>
				</div>,
			);
		}
		return l;
	}, []);

	return (
		<div className={css(style.container)}>
			<div className={css(style.titleContainer)}>
				<input placeholder="Find pokemon" className={css(style.seacrhInput)} />
				<img src={search} alt="search" className={css(style.seacrhImg)} />
			</div>
			<div className={css(style.cardListContainer)}>{list}</div>
		</div>
	);
};

export { CardSearch };
