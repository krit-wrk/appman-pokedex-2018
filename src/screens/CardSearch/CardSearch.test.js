import React from "react";
import { Provider } from "react-redux";
import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import axios from "axios";
import { CardSearch } from "./CardSearch";
import { StyleSheetTestUtils } from "aphrodite";
import { store, myDexAction } from "../../redux";

const getMockData = () => ({
	cards: [
		{
			id: "ex8-98",
			name: "Deoxys ex",
			nationalPokedexNumber: 386,
			imageUrl: "https://images.pokemontcg.io/ex8/98.png",
			imageUrlHiRes: "https://images.pokemontcg.io/ex8/98_hires.png",
			supertype: "Pokémon",
			subtype: "EX",
			ability: {
				name: "Form Change",
				text:
					"Once during your turn (before your attack), you may search your deck for another Deoxys ex and switch it with Deoxys ex. (Any cards attached to Deoxys ex, damage counters, Special Conditions, and effects on it are now on the new Pokémon.) If you do, put Deoxys ex on top of your deck. Shuffle your deck afterward. You can't use more than 1 Form Change Poké-Power each turn.",
				type: "Poké-Power",
			},
			hp: "110",
			retreatCost: ["Colorless", "Colorless"],
			convertedRetreatCost: 2,
			number: "98",
			artist: "Mitsuhiro Arita",
			rarity: "Rare Holo EX",
			series: "EX",
			set: "Deoxys",
			setCode: "ex8",
			text: [
				"When Pokémon-ex has been Knocked Out, your opponent takes 2 Prize cards.",
			],
			attacks: [
				{
					cost: ["Psychic", "Colorless", "Colorless"],
					name: "Psychic Burst",
					text:
						"You may discard 2 Energy attached to Deoxys ex. If you do, this attack does 50 damage plus 20 more damage for each Energy attached to the Defending Pokémon.",
					damage: "50+",
					convertedEnergyCost: 3,
				},
			],
			weaknesses: [
				{
					type: "Psychic",
					value: "×2",
				},
			],
			type: "Psychic",
		},
		{
			id: "dp6-90",
			name: "Cubone",
			nationalPokedexNumber: 104,
			imageUrl: "https://images.pokemontcg.io/dp6/90.png",
			imageUrlHiRes: "https://images.pokemontcg.io/dp6/90_hires.png",
			supertype: "Pokémon",
			subtype: "Basic",
			hp: "60",
			retreatCost: ["Colorless"],
			convertedRetreatCost: 1,
			number: "90",
			artist: "Kagemaru Himeno",
			rarity: "Common",
			series: "Diamond & Pearl",
			set: "Legends Awakened",
			setCode: "dp6",
			attacks: [
				{
					cost: ["Colorless"],
					name: "Headbutt",
					text: "",
					damage: "10",
					convertedEnergyCost: 1,
				},
				{
					cost: ["Fighting", "Colorless"],
					name: "Bonemerang",
					text:
						"Flip 2 coins. This attack does 20 damage times the number of heads.",
					damage: "20×",
					convertedEnergyCost: 2,
				},
			],
			resistances: [
				{
					type: "Lightning",
					value: "-20",
				},
			],
			weaknesses: [
				{
					type: "Water",
					value: "+10",
				},
			],
			type: "Fighting",
		},
	],
});

describe("CardSearch", () => {
	beforeEach(() => {
		StyleSheetTestUtils.suppressStyleInjection();
		store.dispatch(myDexAction.reset());
	});
	afterEach(() => {
		StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
	});
	it("renders without crashing", async () => {
		jest.spyOn(axios, "get").mockResolvedValue({
			data: { cards: [] },
		});
		const { container, getByText } = render(
			<Provider store={store}>
				<CardSearch />
			</Provider>,
		);
		expect(container).toMatchSnapshot();
		expect(getByText("Searching...")).toBeInTheDocument();
		await waitFor(async () =>
			expect(getByText("No search results found")).toBeInTheDocument(),
		);
	});
	it("add ok", async () => {
		jest.spyOn(axios, "get").mockResolvedValue({
			data: getMockData(),
		});

		const { container, getByText, queryByText } = render(
			<Provider store={store}>
				<CardSearch />
			</Provider>,
		);
		expect(container).toMatchSnapshot();
		await waitFor(async () =>
			expect(getByText("DEOXYS EX")).toBeInTheDocument(),
		);
		fireEvent.mouseEnter(getByText("DEOXYS EX"));
		const add = getByText("Add");
		expect(add).toBeInTheDocument();
		expect(container).toMatchSnapshot();
		fireEvent.click(add);
		expect(queryByText("DEOXYS EX")).not.toBeInTheDocument();
		expect(container).toMatchSnapshot();
		expect(store.getState().myDex.myDex.length).toEqual(1);
		expect(store.getState().myDex.myDexDisplay.length).toEqual(1);
	});
});
