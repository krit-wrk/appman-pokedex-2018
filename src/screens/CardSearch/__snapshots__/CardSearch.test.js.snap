// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSearch add ok 1`] = `
<div>
  <div
    class="container_14qfrn4"
  >
    <div
      class="titleContainer_1ygy1t"
    >
      <input
        class="seacrhInput_1ys00gq"
        placeholder="Find pokemon"
        value=""
      />
      <img
        alt="search"
        class="seacrhImg_19gehg2"
        src="search.png"
      />
    </div>
    <div
      class="EmptyContainer_ge2zda"
    >
      <div
        class="EmptyText_sxmy1e"
      >
        Searching...
      </div>
    </div>
  </div>
</div>
`;

exports[`CardSearch add ok 2`] = `
<div>
  <div
    class="container_14qfrn4"
  >
    <div
      class="titleContainer_1ygy1t"
    >
      <input
        class="seacrhInput_1ys00gq"
        placeholder="Find pokemon"
        value=""
      />
      <img
        alt="search"
        class="seacrhImg_19gehg2"
        src="search.png"
      />
    </div>
    <div
      class="cardListContainer_1m8x2n3"
    >
      <div
        class="card_1dgqvry"
      >
        <div
          class="container_b4ahsr"
        >
          <div
            class="imageContainer_1g6vvub"
          >
            <img
              alt="Deoxys ex"
              class="image_5hddva"
              src="https://images.pokemontcg.io/ex8/98.png"
            />
          </div>
          <div
            class="desc_19c633u"
          >
            <div
              class="titleContainer_1ryugy2"
            >
              <div
                class="name_4p276u"
              >
                DEOXYS EX
              </div>
              <div>
                <div
                  class="addButton_5gyw2"
                >
                  Add
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                HP
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                STR
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_1hfu0v"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                WEAK
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="cuteContainer_zz3svu"
            >
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="card_1dgqvry"
      >
        <div
          class="container_b4ahsr"
        >
          <div
            class="imageContainer_1g6vvub"
          >
            <img
              alt="Cubone"
              class="image_5hddva"
              src="https://images.pokemontcg.io/dp6/90.png"
            />
          </div>
          <div
            class="desc_19c633u"
          >
            <div
              class="titleContainer_1ryugy2"
            >
              <div
                class="name_4p276u"
              >
                CUBONE
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                HP
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_1tlr5t6"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                STR
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                WEAK
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="cuteContainer_zz3svu"
            >
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CardSearch add ok 3`] = `
<div>
  <div
    class="container_14qfrn4"
  >
    <div
      class="titleContainer_1ygy1t"
    >
      <input
        class="seacrhInput_1ys00gq"
        placeholder="Find pokemon"
        value=""
      />
      <img
        alt="search"
        class="seacrhImg_19gehg2"
        src="search.png"
      />
    </div>
    <div
      class="cardListContainer_1m8x2n3"
    >
      <div
        class="card_1dgqvry"
      >
        <div
          class="container_b4ahsr"
        >
          <div
            class="imageContainer_1g6vvub"
          >
            <img
              alt="Cubone"
              class="image_5hddva"
              src="https://images.pokemontcg.io/dp6/90.png"
            />
          </div>
          <div
            class="desc_19c633u"
          >
            <div
              class="titleContainer_1ryugy2"
            >
              <div
                class="name_4p276u"
              >
                CUBONE
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                HP
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_1tlr5t6"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                STR
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="descContainer_14nb90c"
            >
              <div
                class="descText_1r7w1yk"
              >
                WEAK
              </div>
              <div
                class="descBar_2jj7wg"
              >
                <div
                  class="container_tze1ke"
                >
                  <div
                    class="bar_1o8e29j-o_O-bar_zdxht7"
                  />
                </div>
              </div>
            </div>
            <div
              class="cuteContainer_zz3svu"
            >
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
              <img
                alt="cute"
                class="cuteImage_1pmj90s"
                src="cute.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CardSearch renders without crashing 1`] = `
<div>
  <div
    class="container_14qfrn4"
  >
    <div
      class="titleContainer_1ygy1t"
    >
      <input
        class="seacrhInput_1ys00gq"
        placeholder="Find pokemon"
        value=""
      />
      <img
        alt="search"
        class="seacrhImg_19gehg2"
        src="search.png"
      />
    </div>
    <div
      class="EmptyContainer_ge2zda"
    >
      <div
        class="EmptyText_sxmy1e"
      >
        Searching...
      </div>
    </div>
  </div>
</div>
`;
